connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaOffsetBackingStore.java

```diff
@@ -141,24 +145,13 @@ private static String noClientId() {
     protected KafkaBasedLog<byte[], byte[]> offsetLog;
     // Visible for testing
     final HashMap<ByteBuffer, ByteBuffer> data = new HashMap<>();
+    private final Map<String, Set<Map<String, Object>>> connectorPartitions = new HashMap<>();
+    private Converter keyConverter;
     private final Supplier<TopicAdmin> topicAdminSupplier;
     private final Supplier<String> clientIdBase;
     private SharedTopicAdmin ownTopicAdmin;
     protected boolean exactlyOnce;
 
-    /**
-     * Create an {@link OffsetBackingStore} backed by a Kafka topic. This constructor will cause the
-     * store to instantiate and close its own {@link TopicAdmin} during {@link #configure(WorkerConfig)}
-     * and {@link #stop()}, respectively.
-     *
-     * @deprecated use {@link #KafkaOffsetBackingStore(Supplier, Supplier)} instead
-     */
-    @Deprecated
-    public KafkaOffsetBackingStore() {
```

Comment:
Sorry for making noise on this PR. out of curiosity, should we remove deprecated constructors from `KafkaStatusBackingStore` and `KafkaConfigBackingStore` too? not sure whether those internal classes need the deprecation cycle?

Reply:
@chia7712 I've raised this PR to remove the other two deprecated constructors - https://github.com/apache/kafka/pull/15865 (and also added some context in the PR on why deprecation was introduced rather than removal even though these classes are not in the public API).